services/replicator/metadataReconciler.go

```diff
@@ -352,7 +352,15 @@ func (r *metadataReconciler) reconcileDestExtentMetadata() error {
 					continue
 				}
 
-				if err = r.reconcileDestExtent(dest.GetDestinationUUID(), localExtents, remoteExtents, zoneConfig.GetZone()); err != nil {
+				// only need to reconcile for extents that are originated from the remote zone
+				filteredLocalExtents := make(map[string]shared.ExtentStatus)
```

Comment:
1. can we move this inside getAllDestExtentInCurrentZone?
No because the filtering is done per zone. Let's say we have zone a,b,c. getAllDestExtentInCurrentZone return extents a1,a2,b1,b2,c1,c2. If current zone is a, remote zone is b, then we only want to have b1,b2 as filteredLocalExtents

2. does the same problem exist in getAllDestExtentInRemoteZone
getAllDestExtentInRemoteZone we filtered on local extent only: https://github.com/uber/cherami-server/blob/master/services/replicator/metadataReconciler.go#L377